src/plan_expand_hypertable.c

```diff
@@ -70,6 +72,124 @@ is_chunk_exclusion_func(Expr *node)
 	return false;
 }
 
+static bool
+is_time_bucket_function(Expr *node)
+{
+	if (IsA(node, FuncExpr) &&
+		strncmp(get_func_name(castNode(FuncExpr, node)->funcid), "time_bucket", NAMEDATALEN) == 0)
+		return true;
+
+	return false;
+}
+
+/*
+ * Transform time_bucket calls of the following form in WHERE clause:
+ *
+ * time_bucket(width, column) > value
+ * column > value - width
+ *
+ * time_bucket(width, column) < value
+ * column < value + width
+ *
+ * This type of WHERE clause is used when querying continuous aggregation
+ * views.
+ */
+static OpExpr *
```

Comment:
would you add a note saying `this function only works with the two argument time_bucket, it's the callers responsibility to ensure that it is only called with such` or add checks of such into the function?

Reply:
done